### DISCUSSION TITLE
ClickHouse Auto-Replication

### DISCUSSION THREAD

**Discussion by karthik-thiyagarajan** [ClickHouse/ClickHouse#67868] [Q&A]
If i have a 3 shard 3 replica setup of clickhouse pods deployed in k8s cluster, and i see that the distribution of my pod is uneven across the availability zones (1 in zone-a, 4 in zone-b, 4 in zone-c) due to my misconfiguration of clickhouse operator config "maxPodsPerNode". Now, if i delete the PVC of any 2 replicas in each shard and ensure that only one replica in each shard is available - and if i redeploy clickhouse - will it automatically cascade the data from that available shard to the new replicas eventually ? Is it a right approach to proceed ?

**Comment by den-crane** [comment]
Operator is a third-party product. Ask them https://github.com/Altinity/clickhouse-operator/issues

**Reply by karthik-thiyagarajan** [reply]
But this is a question on auto-replication when the pvcs of two nodes are lost or deleted and redeployed. Is it not going to replicate the shard's data to other new replicas created ?

**Reply by den-crane** [reply]
Replicated tables maintain data consistency. 
If you create or recreate replica it will fetch missing data automatically.

**Reply by karthik-thiyagarajan** [reply]
Thank you. I will try this and let you know if it works.
